batch/batch/disk.py

```diff
@@ -0,0 +1,120 @@
+import uuid
+import logging
+
+from hailtop.utils import sleep_and_backoff, check_shell
+
+log = logging.getLogger('disk')
+
+
+class Disk:
+    @staticmethod
+    async def delete_disk(compute_client, name, zone):
+        log.info(f'deleting disk {name}')
+        resp = None
+        delay = 0.2
+        delete_uuid = str(uuid.uuid4())
+        while resp is None or resp['status'] != 'DONE':  # pylint: disable=unsubscriptable-object
+            resp = await compute_client.delete(
+                f'/zones/{zone}/disks/{name}',
+                params={'requestId': delete_uuid})
+            delay = await sleep_and_backoff(delay)
```

Comment:
This adds an unnecessary sleep after a successful delete, maybe this instead:

```
delay = 0.2
delete_uuid = str(uuid.uuid4())
while True:
    resp = await compute_client.delete(
        f'/zones/{zone}/disks/{name}',
        params={'requestId': delete_uuid})
    if resp['status'] == 'DONE':
        break
    delay = await sleep_and_backoff(delay)
```

Reply:
Good catch!